lib/modules/dosomething/dosomething_campaign/js/campaign_node_form.js

```diff
@@ -1,33 +1,47 @@
 (function ($) {
   $(document).ready(function(){
 
-    // List all id's that should only display when Campaign type == 'campaign'.
+    // Titles of all elements that should only display when Campaign type == 'campaign'.
+    var campaignTitles = ["Know It", "Plan It", "Prove It"];
+
+    // List all elements that should only display when Campaign type == 'campaign'.
     var campaignSections = [];
-    campaignSections.push("#node_campaign_form_group_prep_it");
-    campaignSections.push("#node_campaign_form_group_do_it");
-    campaignSections.push("#node_campaign_form_group_report_back");
-    campaignSections.push("#node_campaign_form_group_pitch_additional_info");
 
-    // If SMS Game, hide the campaignSections.
-    if ($('#edit-field-campaign-type-und-sms-game').is(':checked')) { 
-      for (var i = 0; i < campaignSections.length; i++) {
-        $(campaignSections[i]).hide();
+    $('.fieldset-title').each(function(index, value) {
+      // Get the title of the fieldset and strip out unneeded text
+      var title = $(value).text().replace("Show", "").trim();
+      //Check if it's a campaign only title
+      if (campaignTitles.indexOf(title) != -1) {
+        campaignSections.push($(value).closest('.form-wrapper'));
       }
+    });
+
+    // If SMS Game, hide the campaignSections.
+    if ($('#edit-field-campaign-type-und-sms-game').is(':checked')) {
+      toggleSections(false);
     }
-    
+
     // Listen for changes to the campaign type field.
     $('#edit-field-campaign-type-und-campaign, #edit-field-campaign-type-und-sms-game').click(function() {
-      if ($('#edit-field-campaign-type-und-campaign').is(':checked')) { 
-        for (var i = 0; i < campaignSections.length; i++) {
-          $(campaignSections[i]).show();
-        }
+      if ($('#edit-field-campaign-type-und-campaign').is(':checked')) {
+        toggleSections(true);
       }
-      if ($('#edit-field-campaign-type-und-sms-game').is(':checked')) { 
-        for (var i = 0; i < campaignSections.length; i++) {
-          $(campaignSections[i]).hide();
-        }
+      if ($('#edit-field-campaign-type-und-sms-game').is(':checked')) {
+        toggleSections(false);
       }
     });
 
+    // Toggle the visibility of field sets in campaignSelections
+    function toggleSections(show) {
+      campaignSections.forEach(function($element, index, array) {
+        if (show) {
+          $element.show();
```

Comment:
I feel like this could be simplified into $element[state](); --- instead of a show parameter people would pass in a string containing 'show' or 'hide', but not sure if it really matters